modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js

```diff
@@ -258,6 +277,13 @@ class OpencastToPaellaConverter {
         }
       }
     });
+    // HLS-VOD Where there's a master HLS index, remove all non-master HLS sources
+    if (hasAdaptiveMasterTrack && currentStream.sources.hls) {
+      var filteredHls = currentStream.sources.hls.filter(track => track.master);
+      currentStream.sources.hls = filteredHls;
+      /****  NOTE: This is a hack to force it to play HLS as a demo, probably wrong, Please fix ****/
+      delete currentStream.sources.mp4;
+    }
 
```

Comment:
@phoebem That delete line is not in DCE's deploy. Is it needed here because the version of OC develop Paella prefers mp4 to hls source?

Reply:
@phoebem, @karendolan In the default configuration mp4 take priority over HLS. You can change this in the `config.json`. The order in which the chargers appear in the list defines the priority-

https://github.com/opencast/opencast/blob/develop/etc/ui-config/mh_default_org/paella/config.json#L31-L76

Reply:
> @phoebem, @karendolan In the default configuration mp4 take priority over HLS. You can change this in the `config.json`. The order in which the loaders appear in the list defines the priority.
> 
@miesgre thanks! If we reverse the order in the config to put HLSVideoFactory before Html5VideoFactory, is it true that browsers that do no support HLS, will still will fall back to the mp4 sources?

Reply:
Yes, Paella uses the first method the browser supports. We are using it in our installation.